src/vm/moar/runner/main.c

```diff
@@ -349,7 +347,13 @@ int wmain(int argc, wchar_t *wargv[])
     }
 #endif
 
-    MVM_vm_exit(instance);
+    if (full_cleanup) {
+        MVM_vm_destroy_instance(instance);
+        return EXIT_SUCCESS;
+    }
+    else {
+        MVM_vm_exit(instance);
+    }
```

Comment:
While it doesn't block merging this at all, note that the `free` lines below are unreachable, due to the fact that the first branch `return`s and the second calls `exit`. This isn't a change from before, but is maybe worth looking at.

Reply:
I fixed this in 34162e967161f34755fb .